### PR TITLE
Add global coverage and add provider admin names

### PR DESCRIPTION
**Pull request by b-j-mills**
I've added global coverage to several of the pipelines (just missing population, operational presence, and poverty rate).  I also added in the provider admin names and p-coded them where possible.  Where p-coding wasn't possible or just very complicated, like the IPC data, I p-coded to a lower admin level so all data rows should be present in the database.

I also ran a test database export and it's working as expected.

### CODE REVIEW

File: src/hapi/pipelines/database/food_security.py [hunk 1]
```diff
@@ -335,6 +335,12 @@ def populate(self) -> None:
                 countryiso3 = row["Country"]
                 if countryiso3 not in self._configuration["HAPI_countries"]:
                     continue
+                provider_admin1_name = row.get("Level 1")
+                if not provider_admin1_name:
+                    provider_admin1_name = ""
+                provider_admin2_name = row.get("Area")
+                if not provider_admin2_name:
+                    provider_admin2_name = ""
```
Comment (mcarans) [inline src/hapi/pipelines/database/food_security.py:343]: Can just do row.get("Level 1", "")

File: src/hapi/pipelines/database/conflict_event.py [hunk 2]
```diff
@@ -73,11 +77,22 @@ def populate(self) -> None:
                             time_period_range = parse_date_range(
                                 f"{month} {year}", "%B %Y"
                             )
+                            provider_admin1_name = ""
+                            provider_admin2_name = ""
+                            if admin_level == "admintwo":
```
Comment (mcarans) [inline src/hapi/pipelines/database/conflict_event.py:82]: Suggest 
if admin_level...  
else:
    provider_admin1_name = "" 
    provider_admin2_name = ""

File: src/hapi/pipelines/database/idps.py [hunk 3]
```diff
@@ -81,9 +81,21 @@ def populate(self) -> None:
...
+                    provider_admin1_name = ""
+                    provider_admin2_name = ""
+                    if "#adm1+name" in hxl_tags:
+                        provider_admin1_name = row[
+                            hxl_tags.index("#adm1+name")
+                        ]
+                    if "#adm2+name" in hxl_tags:
+                        provider_admin2_name = row[
+                            hxl_tags.index("#adm2+name")
+                        ]
```
Comment (mcarans) [inline src/hapi/pipelines/database/idps.py:93]: Again, suggest using if else and set provider_admin1/2_name = "" in the else part. I notice similar code in operational_presence - could this be a utility function?

File: src/hapi/pipelines/database/population.py [hunk 4]
```diff
@@ -54,18 +56,33 @@ def populate(self) -> None:
...
+                        provider_admin1_name = ""
+                        provider_admin2_name = ""
+                        if "#adm1+name" in hxl_tags:
+                            provider_admin1_name = values[
+                                hxl_tags.index("#adm1+name")
+                            ][admin_code]
+                        if "#adm2+name" in hxl_tags:
+                            provider_admin2_name = values[
+                                hxl_tags.index("#adm2+name")
+                            ][admin_code]
```
Comment (mcarans) [inline src/hapi/pipelines/database/population.py:75]: see if else comment for idps

File: src/hapi/pipelines/database/operational_presence.py [hunk 5]
```diff
@@ -153,6 +153,22 @@ def populate(self) -> None:
...
+                                hxl_tags.index("#adm1+name")
+                            ][admin_code][i]
+                        if "#adm2+name" in hxl_tags:
+                            provider_admin2_name = values[
+                                hxl_tags.index("#adm2+name")
+                            ][admin_code][i]
+                        if provider_admin1_name is None:
+                            provider_admin1_name = ""
+                        if provider_admin2_name is None:
+                            provider_admin2_name = ""
```
Comment (mcarans) [inline src/hapi/pipelines/database/operational_presence.py:170]: See if then idps comment

**Review by mcarans — APPROVED** [review]
Great work Briar! The only small change is suggest making a function for the common code to set provider_admin1/2_name and use if/else

**Comment by b-j-mills** [comment]
Thanks @mcarans!  I've added a utility function to get the provider names.

**Pull request merged**